### PR TITLE
UploadFromFileAsync adds the byte order mark (BOM) unicode character

### PR DESCRIPTION
**Pull request by alrod**
This is causing Json.Parse error

### CODE REVIEW

**Comment by fabiocav** [comment]
Which code was failing to parse the file contents? That code would still fail with the current changes if the original file has a BOM, which is possible.

**Comment by fabiocav** [comment]
What code was failing to parse the contents? We need to look at that as we need to be able to cope with blobs containing a BOM

**Comment by alrod** [comment]
The code was:
```
        private static TResult ResolveSerializerAndRun<TResult>(string secretsJson, Type secretsType, Func<IScriptSecretSerializer, JObject, TResult> func)
        {
            JObject secrets = JObject.Parse(secretsJson);
```

Character BOM code is 65279.

File: src/WebJobs.Script.WebHost/Security/KeyManagement/BlobStorageSecretsMigrationRepository.cs [hunk 1]
```diff
@@ -132,8 +132,21 @@ public void Dispose()
                 {
                     string blobName = Path.GetFileName(file);
                     CloudBlockBlob cloudBlockBlob = blobStorageSecretsRepository.BlobContainer.GetBlockBlobReference(blobStorageSecretsRepository.SecretsBlobPath + "/" + blobName);
-                    Task copyTask = cloudBlockBlob.UploadFromFileAsync(file);
+
+                    string contents = File.ReadAllText(file);
+                    Task copyTask = cloudBlockBlob.UploadTextAsync(contents);
                     copyTasks.Add(copyTask);
+
+                    //Task copyTask = new Task(async () =>
```
Comment (fabiocav) [inline src/WebJobs.Script.WebHost/Security/KeyManagement/BlobStorageSecretsMigrationRepository.cs]: I assume the commented out code will be removed, right?

**Comment by fabiocav** [comment]
We deal with this in a few different places in the codebase (and in some places, ensure the BOM is removed, or ensure proper encoding configuration, when appropriate). You'll find a few utility helpers do deal with that.

**Pull request merged**